catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -377,9 +382,57 @@ protected SolrQuery postAdapt(QueryRequest request, SolrFilterDelegate filterDel
 
         setSortProperty(request, query, filterDelegate);
 
+        filterAttributes(request, query);
+
         return query;
     }
 
+    private void filterAttributes(QueryRequest request, SolrQuery query) {
+        if (!request.hasProperties() || !request.containsPropertyName(EXCLUDE_ATTRIBUTES)
+                || !(request.getPropertyValue(EXCLUDE_ATTRIBUTES) instanceof Set) ||
+                "true".equals(System.getProperty("solr.client.filterAttributes.disable"))) {
```

Comment:
✏️ This condition (the system property check) seems a little awkward here with the property existence check and instanceof check. Maybe we could pull it out into its' own if statement above this one, that way we leave the logic relating to the exclude attributes grouped together and the system property check on its own.